### ISSUE TITLE
nordpy on the command line

### ISSUE THREAD

**Issue by 5t4cktr4c3** [morpheusthewhite/nordpy#46]
Hi,
currently I'm trying to automate the connection to the NordVPN network. Using Google I found your project, which seemed quite useful, so I installed it on my Raspberry Pi (Raspbian Lite). However, on trying to start nordpy it fails and exits with the message that the tkinter package is not present. And there's my question: I want to use nordpy exclusively on the command line, so is it possible to install nordpy without the GUI?

Thanks!

**Comment by morpheusthewhite** [comment]
Currently that is not possible, since in order to start the program from command line you need to connect at least once by using the gui.

This is necessary both to store your credentials and preferences.

**Comment by 5t4cktr4c3** [comment]
Can you consider implementing asking for those credentials and preferences on the installation process?

**Comment by morpheusthewhite** [comment]
Yes, I'll this issue open and see if someone else is interested and/or may want to contribute

**Comment by morpheusthewhite** [comment]
Implementing command line supports means:
- [ ] ask and store credentials from command line
- [ ] ask and store preferences to be used in next command line runs

Optionally also
- [ ] provide arguments to eventually choose preferences (when stored ones needs to be overwritten)

**Comment by 5t4cktr4c3** [comment]
I'd like to contribute! However, I think I need some time to get into that. Moreover, I haven't contributed to any other repository on GitHub yet, so I'm not familiar with that system.

**Comment by morpheusthewhite** [comment]
> I'd like to contribute! However, I think I need some time to get into that. Moreover, I haven't contributed to any other repository on GitHub yet, so I'm not familiar with that system.

I you decide to do so, you can easily find some resources for the web (regarding git and github, which are very simple for what you need) as well as for python (if you need, I suggest the official docs).

Strictly regarding this project, if you need hints about what you need to do don't exitate writing here

**Comment by 5t4cktr4c3** [comment]
I looked into the code in the last few days. I saw that tkinter is used very widely to control the application.
What do you think about restructuring the whole project, so that the base functionality is served via CLI and Python tools (for example fetching the ovpn.zip via the requests-module) and that the GIU is based upon those CLI?
Also, I would welcome supporting macOS as well.

**Comment by morpheusthewhite** [comment]
I don't think you need to restructure the whole project. Yes, tkinter is very important in the application but some part of it are already tkinter-agnostic, like everything related to connection handling and settings too; on the other hand some other parts like credentials are coupled with gui. You can only to focus on that part, I don't think the rest of the code needs to be changed.

Regarding ovpn.zip as you probably saw the initial population is done via install.sh, which simplifies some things wrt to python since they are mostly operations on the filesystem and dependency installation. (note: python handles ovpn.zip update via requests module, but that function is called in very rare occasions).

I cannot test on macOS so I am not clearly not able to garantee full support for it. Also it wouldn't surprise me it this would not work since it relies on some things which may be different on macOS (I actually don't know, I never used it) like system DNS and iptables.
